hasura/graphql-engine
# add spatial predicates for geography columns (close #1674)

### Description

This PR adds support for PostGIS spatial predicates on `geography` columns. The predicates are [`_st_d_within`](https://postgis.net/docs/ST_DWithin.html) and [`_st_intersects`](https://postgis.net/docs/ST_Intersects.html).

### Affected components 


- Server (adds required types)
- Console (Permission builder needs update)
- Tests
- Docs

### Related Issues
#1674 

### Solution and Design
The comparison expression input type generating function is extended to support `geography` the same way `geometry` is supported.

### Steps to test and verify
Start hasura on Postgres with PostGIS enabled, create a table with `geography` and `geometry` columns:
```sql
create table test (
  id serial primary key,
  geometry_col geometry,
  geography_col geography
)
```
Using console GraphiQL Docs tab, verify that the `geography_comparison_exp` contains `_st_d_within` and `_st_intersects`.

### Limitations, known bugs & workarounds
NA
